### PR TITLE
Instrument vertx client directly

### LINKED ISSUE #310: Vertx client instrumentation is broken in 1.1
Fix will require instrumenting vertx client directly (instead of Netty as we do now).

Disabled test: https://github.com/hypertrace/javaagent/pull/308/files#diff-7bf0cebf0d08303f88f759ed4f34c318c1dc246cf8bb787a313a45cdb73b6576R33

### PR DESCRIPTION
**Pull request by pavolloffay**
Resolves #310

### CODE REVIEW

File: instrumentation/vertx-web-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/vertx/HttpRequestInstrumentation.java [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+          InstrumentationContext.get(HttpClientRequest.class, Contexts.class).get(request);
+      if (contexts == null) {
+        return;
+      }
+      Span span = Span.fromContext(contexts.context);
+
+      BoundedCharArrayWriter buffer =
+          InstrumentationContext.get(MultiMap.class, BoundedCharArrayWriter.class)
+              .get(request.headers());
+      if (buffer != null) {
```
Comment (shashank11p) [inline instrumentation/vertx-web-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/vertx/HttpRequestInstrumentation.java:106]: Dont we want to check if 
```
(HypertraceConfig.get().getDataCapture().getHttpBody().getRequest().getValue()
          && ContentTypeUtils.shouldCapture(contentType))

File: instrumentation/vertx-web-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/vertx/HttpRequestInstrumentation.java [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+          && ContentTypeUtils.shouldCapture(contentType)) {
+
+        ContextStore<MultiMap, BoundedCharArrayWriter> contextStore =
+            InstrumentationContext.get(MultiMap.class, BoundedCharArrayWriter.class);
+        BoundedCharArrayWriter buffer = contextStore.get(request.headers());
+        if (buffer == null) {
+          buffer = BoundedBuffersFactory.createWriter();
+          contextStore.put(request.headers(), buffer);
+        }
+        buffer.write(chunk);
```
Comment (shashank11p) [inline instrumentation/vertx-web-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/vertx/HttpRequestInstrumentation.java:220]: Why are we calling this here and in EndRequestAdvice both?

File: instrumentation/vertx-web-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/vertx/ResponseBodyWrappingHandler.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  public ResponseBodyWrappingHandler(Handler<Buffer> wrapped, Span span) {
+    this.wrapped = wrapped;
+    this.span = span;
+  }
+
+  @Override
+  public void handle(Buffer event) {
+    String responseBody = event.getString(0, event.length());
+    if (span.isRecording()) {
+      span.setAttribute(HypertraceSemanticAttributes.HTTP_RESPONSE_BODY, responseBody);
```
Comment (shashank11p) [inline instrumentation/vertx-web-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/vertx/ResponseBodyWrappingHandler.java:44]: Should we also check if we should capture response body?

File: instrumentation/vertx-web-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/vertx/HttpRequestInstrumentation.java (hunk 2 above)
Comment (pavolloffay) [reply instrumentation/vertx-web-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/vertx/HttpRequestInstrumentation.java:220]: yes, this writes a chunk and in the end method it as well writes a chunk and sends the request.

File: instrumentation/vertx-web-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/vertx/HttpRequestInstrumentation.java (hunk 1 above)
Comment (pavolloffay) [reply instrumentation/vertx-web-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/vertx/HttpRequestInstrumentation.java:106]: we don't have to check it here bc the buffer is null if the body shouldn't be captured.

File: instrumentation/vertx-web-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/vertx/ResponseBodyWrappingHandler.java (hunk 3 above)
Comment (pavolloffay) [reply instrumentation/vertx-web-3.0/src/main/java/io/opentelemetry/javaagent/instrumentation/hypertrace/vertx/ResponseBodyWrappingHandler.java:44]: not needed as this is reached only after the check from the previous instrumentation point.

**Pull request merged**